com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -12,282 +13,181 @@ namespace MLAPI.Prototyping
+                    isDirty |= m_NetworkState.Value.Rotation != transform.rotation;
+                    isDirty |= m_NetworkState.Value.Scale != transform.lossyScale;
                 }
+
+                return isDirty;
             }
         }
 
-        private Quaternion m_CurrentRotation
+        private void UpdateNetworkState()
         {
-            get { return m_UseLocal.Value ? m_Transform.localRotation : m_Transform.rotation; }
-            set
+            m_NetworkState.Value.InLocalSpace = InLocalSpace;
+            if (InLocalSpace)
             {
-                if (m_UseLocal.Value)
-                {
-                    m_Transform.localRotation = value;
-                }
-                else
-                {
-                    m_Transform.rotation = value;
-                }
+                m_NetworkState.Value.Position = transform.localPosition;
+                m_NetworkState.Value.Rotation = transform.localRotation;
+                m_NetworkState.Value.Scale = transform.localScale;
             }
-        }
-
-        private Vector3 m_CurrentScale
-        {
-            get { return m_UseLocal.Value ? m_Transform.localScale : m_Transform.lossyScale; }
-            set
+            else
             {
-                if (m_UseLocal.Value)
-                {
-                    m_Transform.localScale = value;
-                }
-                else
-                {
-                    SetWorldScale(value);
-                }
+                m_NetworkState.Value.Position = transform.position;
+                m_NetworkState.Value.Rotation = transform.rotation;
+                m_NetworkState.Value.Scale = transform.lossyScale;
             }
-        }
-
-        private Vector3 m_OldPosition;
-        private Quaternion m_OldRotation;
-        private Vector3 m_OldScale;
-
-        private NetworkVariable<Vector3>.OnValueChangedDelegate m_PositionChangedDelegate;
-        private NetworkVariable<Quaternion>.OnValueChangedDelegate m_RotationChangedDelegate;
-        private NetworkVariable<Vector3>.OnValueChangedDelegate m_ScaleChangedDelegate;
-
-        // todo really not happy with that one, hopefully we can have a cleaner solution with reparenting.
-        private void SetWorldScale(Vector3 globalScale)
-        {
-            m_Transform.localScale = Vector3.one;
-            var lossyScale = m_Transform.lossyScale;
-            m_Transform.localScale = new Vector3(globalScale.x / lossyScale.x, globalScale.y / lossyScale.y, globalScale.z / lossyScale.z);
-        }
 
-        public bool CanUpdateTransform()
-        {
-            return (IsClient && TransformAuthority == Authority.Client && IsOwner) || (IsServer && TransformAuthority == Authority.Server) || TransformAuthority == Authority.Shared;
+            m_NetworkState.SetDirty(true);
         }
 
-        private void Awake()
+        private void ApplyNetworkState(NetworkState netState)
         {
-            m_Transform = transform;
```

Comment:
A reason why I had m_Transform was this here
https://github.com/JetBrains/resharper-unity/wiki/Avoid-multiple-unnecessary-property-accesses